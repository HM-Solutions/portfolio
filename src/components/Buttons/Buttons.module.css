@import url('https://fonts.googleapis.com/css2?family=Cormorant+SC:wght@300&display=swap');

.BtnNav {
    margin: 0.7rem;
    font-family: 'Cormorant SC', serif;
    font-weight: bold;
}