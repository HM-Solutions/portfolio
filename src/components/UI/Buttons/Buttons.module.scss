.buttonOne:hover {
  background-color: white;
  color: white;
}
