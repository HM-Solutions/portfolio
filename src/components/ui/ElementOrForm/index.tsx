import React from "react";
import useView from "../../../providers/useView";
import { Button } from "../Buttons";
import MessageForm from "../MessageForm";

interface ElementOrFormProps {
  element: any;
  changeViewFunction?: () => void;
  form: any;
  message: string;
}

export default function ElementOrForm(props: ElementOrFormProps) {
  const { view, setView } = useView();
  function tableOrFormView() {
    return !view ? (
      <>
        <div>
          <Button
            onClick={() => {
              setView(true);
              props.changeViewFunction && props.changeViewFunction();
            }}
          >
            CADASTRAR
          </Button>
        </div>
        {props.element}
      </>
    ) : (
      <>
        <Button
          onClick={() => {
            setView(false);
            props.changeViewFunction && props.changeViewFunction();
          }}
        >
          FECHAR FORMUL√ÅRIO
        </Button>
        {props.form}
      </>
    );
  }

  return props.message ? (
    <MessageForm message={props.message} />
  ) : (
    <div style={{ width: "100%", marginTop: "3%", textAlign: "center" }}>
      {tableOrFormView()}
    </div>
  );
}
