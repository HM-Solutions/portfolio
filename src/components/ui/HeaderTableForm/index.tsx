import React from "react";
import { Button } from "../Buttons";

interface HeaderTableFormProps {
  title: string;
}

export default function HeaderTableForm(props: HeaderTableFormProps) {
  return (
    <div
      style={{
        width: "100%",
        display: "flex",
        justifyContent: "left",
        textTransform: "uppercase",
      }}
    >
      <h1>{props.title}</h1>
    </div>
  );
}
