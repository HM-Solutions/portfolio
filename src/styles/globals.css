@import url('https://fonts.googleapis.com/css2?family=Cormorant+SC:wght@300&display=swap');
html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  font-family: 'Cormorant SC', serif;
  box-sizing: border-box;
}

hr {
  margin: 2%;
  color: black;
}

.m-2 {
  margin: 0.7rem;
}

.Title {
  font-family: 'Cormorant SC', serif;
  font-weight: bold;
  color: black;
}

.bold {
  font-family: 'Cormorant SC', serif;
  font-weight: bold;
}


