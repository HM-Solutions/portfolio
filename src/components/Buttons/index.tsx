import React from "react";
import Button from '@mui/material/Button';
import styles from './Buttons.module.css';

interface ButtonProps {
    children: any;
    onClick?: any;
    className?: any;
}
export default function ButtonNav(props: ButtonProps){
    return (

        <Button onClick={props.onClick} className={`${props.className} ${styles.BtnNav}`} variant="contained">{props.children}</Button>
    )
}