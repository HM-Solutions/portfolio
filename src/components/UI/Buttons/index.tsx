import React from "react";
import Button from '@mui/material/Button';
import styles from './Buttons.module.css';

interface ButtonProps {
    children: any;
    onClick?: any;
    className?: any;
}
export default function ButtonOne(props: ButtonProps){
    return (

        <Button onClick={props.onClick} sx={{
            width: '100%',
            margin: '0.7rem',
            padding: '2%',
            fontWeight: 'bold',
            fontFamily: 'Cormorant SC',
            backgroundColor: "rgb(24, 24, 24)",
            color: "white",
        }} variant="contained">{props.children}</Button>
    )
}