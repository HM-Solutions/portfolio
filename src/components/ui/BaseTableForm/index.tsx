import React from "react";
import HeaderTableForm from "./HeaderTableForm";
import styles from "./BaseTableForm.module.scss";
import { iBaseTableForm } from "../../../interfaces/iBaseTableForm";

export default function BaseTableForm(props: iBaseTableForm) {
  return (
    <>
      <HeaderTableForm title={props.title} />
      <table className={styles.table}>
        <thead>
          <tr>
            <th>Description</th>
            <th>Edit</th>
          </tr>
        </thead>
        <tbody>{props.children}</tbody>
      </table>
    </>
  );
}
