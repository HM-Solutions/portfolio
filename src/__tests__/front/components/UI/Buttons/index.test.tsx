import { describe, expect, it, jest } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { Button } from "../../../../../components/ui/Buttons";

describe("UI Buttons Test", () => {
  it("<Button/>", async () => {
    render(<Button>Button Default</Button>);
    const button = await screen.findByText("Button Default");
    expect(button).toMatchSnapshot();
  });

  it("<Button/> with onClick", async () => {
    const onClick = jest.fn();
    render(<Button onClick={() => onClick()}>Button OnClick</Button>);
    const button = await screen.findByText("Button OnClick");
    await button.click();
    expect(button).toMatchSnapshot();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
