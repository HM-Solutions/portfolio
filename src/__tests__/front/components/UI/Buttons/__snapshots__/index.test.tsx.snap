// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Buttons Test <Button/> with onClick 1`] = `
<button
  class="buttonDefault undefined"
>
  Button OnClick
</button>
`;
